extensions/wikia/AuthModal/js/AuthComponent.js

```diff
@@ -0,0 +1,51 @@
+define('AuthComponent', function () {
+	'use strict';
+
+	function AuthComponent (rootElement) {
+		this.pages = {
+				login: 'signin?modal=1',
+				facebookConnect: 'signin?method=facebook&modal=1',
+				register: 'register?modal=1',
+				facebookRegister: 'register?method=facebook&modal=1'
+			};
+		this.rootElement = rootElement;
+	}
+
+	AuthComponent.prototype.getUselangParam  = function (language) {
+		if (typeof language !== 'string') {
+			return '';
+		}
+		return '&uselang=' + language;
+	};
+
+	AuthComponent.prototype.open = function (page, callback, language) {
+		if (this.rootElement instanceof HTMLElement) {
+			var authIframe = document.createElement('iframe');
+			authIframe.src = window.location.origin + '/' + page + this.getUselangParam(language);
```

Comment:
This line looks little bit tricky.
```page``` already contains ```?```, that's why ```uselang``` doesn't require it, but uses ```&```. What do you think about separating path and query params and joining them implicitly in the code?